lib/src/stream_extensions.dart

```diff
@@ -54,4 +54,17 @@ extension StreamExtensions<T> on Stream<T> {
     });
     return completer.future;
   }
+
+  /// Returns a view of this stream that eagerly buffers all events until
+  /// `listen()` is called.
+  Stream<T> buffer() {
```

Comment:
This name conflicts with `buffer` from `stream_transform`.

https://pub.dev/documentation/stream_transform/latest/stream_transform/RateLimit/buffer.html

WDYT about `asSingleSubscriptionStream` to contrast with `asBroadcastStream`?


Would it be sensible to also have

```dart
if (!isBroadcast) return this;
```

Reply:
> Would it be sensible to also have

I guess if the goal is specifically to have a listen on streams that might not otherwise ever have one it wouldn't make sense to `return this`.

We could also consider the name `eagerListen` instead of `asSingleSubscriptionStream` to reinforce the intent.

Reply:
I think `eagerListen()` or `bufferUntilListen()` would make sense as names.

Reply:
Or `eagerlyBuffer()`. 
The `bufferUntilListen` seems most descriptive, `eagerListen` sounds like it's _more_ eager than just doing `listen` now, which isn't possible.

Reply:
I like `bufferUntilListen` even if it's a bit verbose.

Reply:
I could get on board with `bufferUntilListen` if it is the consensus choice, but I like that `eagerListen` highlights the focus on the listen that happens now (which I find to be the interesting one), rather than the listen that happens later.

How about `listenAndBuffer()`?

Reply:
The `listenAndBuffer` is technically correct. It's a low-level (low abstraction) name because it describes the *implementation*, not a higher-level concept.
Maybe because there is no higher level concept, this really is a shorthand for listening-and-buffering, which will only be used as a low-level primitive of other things. Users shouldn't need to use it directly.

Reply:
@lrhn - is that a vote for `listenAndBuffer` or do you still prefer `bufferUntilListen`?

I do think `listenAndBuffer` is the least likely to be misread.

Reply:
Let's say it's a vote for `listenAndBuffer`. It's an ugly name, but it's precise and understandable.

I like that `listen` is early, because listening to a stream is a significant action to take. You'd usually not expect a function returning a stream to do *anything* before you listen to that stream.

If the name is missing something, it is what it's behavior is after listening to the returned stream.
But adding more would be too long, so putting that in the documentation is fine.

:+1:

Reply:
@nex3 - can you rename to `listenAndBuffer`? Also update the name in the CHANGELOG.

Reply:
Oops, done!